src/libraries/System.Text.Encoding/tests/Encoding/TranscodingStreamTests.cs

```diff
@@ -336,10 +336,10 @@ private void RunReadTest(Func<Stream, MemoryStream, int> callback)
 
             Assert.Equal(-1, transcodingStream.ReadByte()); // should've reached EOF
 
-            // Now put some invalid data into the inner stream as EOF.
```

Comment:
Could you also add a test that specifically validates the bug as reported? Thanks.

Reply:
That specific scenario test does exist. See lines 359, 549 of this file.